operator/pkg/upgrade/upgrade.go

```diff
@@ -3,95 +3,92 @@ package upgrade
-	upgradeJobConfigMap = "upgrade-job-configmap-%s"
-)
-
 // Upgrade upgrades the embedded cluster to the version specified in the installation.
 // First the k0s cluster is upgraded, then addon charts are upgraded, and finally the installation is unlocked.
-func Upgrade(ctx context.Context, cli client.Client, in *clusterv1beta1.Installation) error {
-	fmt.Printf("Upgrading to version %s\n", in.Spec.Config.Version)
-	err := k0sUpgrade(ctx, cli, in)
-	if err != nil {
-		return fmt.Errorf("k0s upgrade: %w", err)
-	}
+func Upgrade(ctx context.Context, cli client.Client, in *ecv1beta1.Installation) error {
+	slog.Info("Upgrading Embedded Cluster", "version", in.Spec.Config.Version)
 
 	// Augment the installation with data dirs that may not be present in the previous version.
 	// This is important to do ahead of updating the cluster config.
 	// We still cannot update the installation object as the CRDs are not updated yet.
-	in, err = maybeOverrideInstallationDataDirs(ctx, cli, in)
+	in, err := maybeOverrideInstallationDataDirs(ctx, cli, in)
 	if err != nil {
 		return fmt.Errorf("override installation data dirs: %w", err)
 	}
 
+	err = upgradeK0s(ctx, cli, in)
+	if err != nil {
+		return fmt.Errorf("k0s upgrade: %w", err)
+	}
+
 	// We must update the cluster config after we upgrade k0s as it is possible that the schema
 	// between versions has changed. One drawback of this is that the sandbox (pause) image does
 	// not get updated, and possibly others but I cannot confirm this.
-	err = clusterConfigUpdate(ctx, cli, in)
+	err = updateClusterConfig(ctx, cli)
 	if err != nil {
 		return fmt.Errorf("cluster config update: %w", err)
 	}
 
-	fmt.Printf("Updating registry migration status\n")
-	err = registryMigrationStatus(ctx, cli, in)
+	slog.Info("Upgrading addons")
+	err = upgradeAddons(ctx, cli, in)
 	if err != nil {
-		return fmt.Errorf("registry migration status: %w", err)
+		return fmt.Errorf("upgrade addons: %w", err)
 	}
 
-	fmt.Printf("Upgrading addons\n")
-	err = chartUpgrade(ctx, cli, in)
+	slog.Info("Upgrading extensions")
+	err = upgradeExtensions(ctx, cli, in)
 	if err != nil {
-		return fmt.Errorf("chart upgrade: %w", err)
+		return fmt.Errorf("upgrade extensions: %w", err)
 	}
 
-	fmt.Printf("Waiting for operator chart to be ready\n")
-	// wait for the operator chart to be ready
-	err = waitForOperatorChart(ctx, cli, in.Spec.Config.Version)
-	if err != nil {
-		return fmt.Errorf("wait for operator chart: %w", err)
-	}
-
-	fmt.Printf("Re-applying installation\n")
-	// Finally, re-apply the installation as the CRDs are up-to-date.
+	slog.Info("Re-applying installation")
+	// re-apply the installation as the CRDs are up-to-date.
 	err = reApplyInstallation(ctx, cli, in)
 	if err != nil {
 		return fmt.Errorf("unlock installation: %w", err)
 	}
 
 	err = support.CreateHostSupportBundle()
 	if err != nil {
-		logrus.Warnf("Failed to upgrade host support bundle: %v", err)
+		logrus.Warnf("upgrade host support bundle: %v", err)
```

Comment:
Should put this back as it was more readable before

```suggestion
		logrus.Warnf("Failed to upgrade host support bundle: %v", err)
```

Reply:
This should also probably be slog